### ISSUE TITLE
Graphing with X values that are not in order and repeated

### ISSUE THREAD

**Issue by HarshPatel682** [PhilJay/MPAndroidChart#3373]
Hello,

I've been trying to find a way to make a real-time Line Graph with x-values that are not in numerical order. If I try plotting them, the app just crashes. 
Also, in my case, the x-values are repeated so I assume would it override them.
Can someone please help me out ASAP, need to get this done for a project.

Thanks!!

**Comment by z0rbal** [comment]
Yes, I'd really like this too. I have to plot realtime GPS positions, and by definition, x-values can be duplicated and are always not in order - Also an (almost) vertical line produces problems. It would be really good to have an option to plot scatter with lines between successive points  or a linechart in point order rather than x order - even if it's slower. I have managed to use a different dataset when the data changes direction but this is kludgy and can only do for 6 sets...

**Comment by z0rbal** [comment]
found a solution that seems to work (at least for mine):
in LineChartRenderer.java replace (Line 395):
                    mLineBuffer[j++] = e1.getX();
                    mLineBuffer[j++] = e1.getY() * phaseY;

                    if (isDrawSteppedEnabled) {
                        mLineBuffer[j++] = e2.getX();
                        mLineBuffer[j++] = e1.getY() * phaseY;
                        mLineBuffer[j++] = e2.getX();
                        mLineBuffer[j++] = e1.getY() * phaseY;
                    }

                    mLineBuffer[j++] = e2.getX();
                    mLineBuffer[j++] = e2.getY() * phaseY;
with the following (changes point order if e1-x<e2-x):
                    if (e1.getX() >= e2.getX()) { //added
                        mLineBuffer[j++] = e1.getX();
                        mLineBuffer[j++] = e1.getY() * phaseY;

                        if (isDrawSteppedEnabled) {
                            mLineBuffer[j++] = e2.getX();
                            mLineBuffer[j++] = e1.getY() * phaseY;
                            mLineBuffer[j++] = e2.getX();
                            mLineBuffer[j++] = e1.getY() * phaseY;
                        }

                        mLineBuffer[j++] = e2.getX();
                        mLineBuffer[j++] = e2.getY() * phaseY;
                    }else{ //added
                        Object e2D=e2.getData(); // swap data for on click
                        Object e1D=e1.getData();
                        e1.setData(e2D);
                        e2.setData(e1D); // swap data
                        mLineBuffer[j++] = e2.getX();
                        mLineBuffer[j++] = e2.getY() * phaseY;

                        if (isDrawSteppedEnabled) {
                            mLineBuffer[j++] = e1.getX();
                            mLineBuffer[j++] = e2.getY() * phaseY;
                            mLineBuffer[j++] = e1.getX();
                            mLineBuffer[j++] = e2.getY() * phaseY;
                        }

                        mLineBuffer[j++] = e1.getX();
                        mLineBuffer[j++] = e1.getY() * phaseY;
                    }
                }
